.github/workflows-source/ci-pr-only.yml

```diff
@@ -107,3 +107,35 @@ jobs:
         id: pre-commit
         shell: bash
         run: ./gitlab-ci/src/ci-scripts/pre-commit.sh
+
+  dependencies-check:
+    name: Dependency Scan for PR
+    <<: *dind-small-setup
+    timeout-minutes: 60
+    permissions:
+      pull-requests: write
+    env:
+      SHELL_WRAPPER: "/usr/bin/time"
+      CI_MERGE_REQUEST_IID: ${{ github.event.pull_request.number }}
+      CI_PROJECT_PATH: ${{ github.repository }}
+      CI_PIPELINE_ID: ${{ github.run_id }}
+      CI_COMMIT_SHA: ${{ github.sha }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      JIRA_API_TOKEN: ${{ secrets.JIRA_API_TOKEN }}
+      SLACK_PSEC_BOT_OAUTH_TOKEN: ${{ secrets.SLACK_PSEC_BOT_OAUTH_TOKEN }}
+    steps:
+      - <<: *checkout
+      - <<: *before-script
+      - name: Setup python deps
+        id: setup-python-deps
+        shell: bash
+        run: |
+          pip3 install --ignore-installed -r requirements.txt
```

Comment:
Seems a bit off to have pip3 install in every PR. Would it make sense to put this in the `ic-build` image? Or maybe use `PIP_DOWNLOAD_CACHE` in `/cache/pip`?

Reply:
AFAIR, we used to already have it in the `ic-build` image but @cgundy made a PR long back to move this to outside of the image.

Reply:
Yeah we don't have python packages pre-installed in the image and we install it this way for other CI jobs too. We can discuss later if it makes sense to do a refactor but from what I remember we wanted to keep the images as small as possible.

Reply:
See reference here for why we removed them from the docker image: https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/15567

```
Bazel can run into conflicts when the dependencies are pre-installed in the CI image.
```